pkg/steps/run.go

```diff
@@ -56,7 +56,7 @@ func Run(ctx context.Context, graph []*api.StepNode) (*junit.TestSuites, []api.C
 			stepDetails = append(stepDetails, out.stepDetails)
 			if out.err != nil {
 				testCase.FailureOutput = &junit.FailureOutput{Output: out.err.Error()}
-				if out.err != context.Canceled {
+				if out.err.Error() != context.Canceled.Error() {
```

Comment:
If this is about wrapping, using `errors.Is` seems to be the better choice

Reply:
Nice! Thanks, updated.